### PR TITLE
Svyatoslav/google auth

### PR DESCRIPTION
**Pull request by SvyatoslavArtymyvych**

### CODE REVIEW

**Review by nixissmtp — APPROVED** [review]

File: app/controllers/__init__.py [hunk 1]
```diff
@@ -1 +1,4 @@
 # flake8: noqa F401
+from .google_auth import GoogleAuth
+
+google_auth = GoogleAuth()
```
Comment (kipod) [inline app/controllers/__init__.py:4]: EOL on EOF

File: app/controllers/google_auth.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import os
+
+from google_auth_oauthlib.flow import Flow
+
+
+class GoogleAuth:
+    def __init__(self):
+        host = os.environ.get("HOST")
```
Comment (kipod) [inline app/controllers/google_auth.py:8]:
```python
host = os.environ.get("HOST", "localhost")
```

File: app/controllers/google_auth.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+        self.flow = Flow.from_client_config(
+            client_config={
+                "web": {
+                    "client_id": os.environ.get("GOOLE_CLIENT_ID"),
```
Comment (kipod) [inline app/controllers/google_auth.py:13]: Pls always set the default value. E.g. :
```python
"client_id": os.environ.get("GOOLE_CLIENT_ID", "Unknown Google Client ID"),
```

Comment (kipod) [reply app/controllers/google_auth.py:13]: All these constants I prefer define in the config.py module

File: app/templates/base.html [hunk 4]
```diff
@@ -1,127 +1,132 @@
...
+    <title>{{ config.APP_NAME }}</title>
+    <!-- meta -->
+    <meta name="description" content="" />
+    <meta name="author" content="" />
+    <meta name="viewport" content="width=device-width,initial-scale=1" />
+    {% block meta %}{% endblock %}
+    <!-- styles -->
+    <link
+      rel="stylesheet"
+      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
```
Comment (kipod) [inline app/templates/base.html:14]: IMHO better download and put this file to static folder

File: app/templates/base.html [hunk 5]
```diff
@@ -1,127 +1,132 @@
...
-            </li>
-            <li class="list-item"><a href="https://twitter.com" target="_blank" title="twitter"><span
-                  class="fa fa-twitter"></span>Twitter</a>
-            </li>
-            <li class="list-item"><a href="https://instagram.com" target="_blank" title="instagram"><span
-                  class="fa fa-instagram"></span>Instagram</a></li>
-          </ul>
+                {% if current_user.is_authenticated %}
+                <li class="nav-item">
+                  <a href="profile" class="nav-link">Profile</a>
```
Comment (kipod) [inline app/templates/base.html:74]: ```{{url_for("main.profile")}}```

File: app/templates/base.html [hunk 6]
```diff
@@ -1,127 +1,132 @@
...
+                </li>
+                <li class="nav-item">
+                  <a href="logout" class="nav-link">Logout</a>
```
Comment (kipod) [inline app/templates/base.html:77]: ```{{url_for("auth.logout")}}```

File: app/templates/base.html [hunk 7]
```diff
@@ -1,127 +1,132 @@
...
+                </li>
+                {% else %}
+                <li class="nav-item">
+                  <a href="login" class="nav-link">Login</a>
```
Comment (kipod) [inline app/templates/base.html:81]: ```{{url_for("auth.login")}}```

File: app/templates/base.html [hunk 8]
```diff
@@ -1,127 +1,132 @@
...
+    {% endfor %} {% endif %} {% endwith %}
+
+    <!-- Main Content -->
+    {% block content %}{% endblock %}
 
-</html>
\ No newline at end of file
+    <!-- scripts -->
+    <script
+      src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
```
Comment (kipod) [inline app/templates/base.html:112]: IMHO better download and put this file to static folder

File: app/views/auth.py [hunk 9]
```diff
@@ -1,46 +1,66 @@
...
+    token_request = google.auth.transport.requests.Request(session=cached_session)
+
+    id_info = id_token.verify_oauth2_token(
+        id_token=credentials._id_token,
+        request=token_request,
+        audience=credentials.client_id
+    )
+
+    session["google_id"] = id_info.get("sub")
+    session["name"] = id_info.get("name")
```
Comment (kipod) [inline app/views/auth.py:43]: Are u sure that these values need to store in the Flask session?

File: app/views/auth.py [hunk 10]
```diff
@@ -1,46 +1,66 @@
...
-    form = LoginForm(request.form)
-    if form.validate_on_submit():
-        user = User.authenticate(form.user_id.data, form.password.data)
-        if user is not None:
-            login_user(user)
-            flash("Login successful.", "success")
-            return redirect(url_for("main.index"))
-        flash("Wrong user ID or password.", "danger")
-    return render_template("auth/login.html", form=form)
+    return redirect("/profile")
```
Comment (kipod) [inline app/views/auth.py:57]: use url_for()

**Review by kipod — COMMENTED** [review]
Thank you!

1. what problem is to write tests for google auth?
2. Enable run pytest in GitHub Actions

**Review by kipod — APPROVED** [review]
Ok, thank you

**Pull request merged**